Add gprimeltecan, fortuna3g, fortunaltezt

stephan-gh opened the pull request:
This is a minor cleanup of #227. The main change is that more hardware is placed in the common `gprime.dtsi`, e.g. the sensors, parts of NFC and the pwm-backlight. The hardware is then selected in the variant-specific dts parts as needed.

@wonderfulShrineMaidenOfParadise, @SiddharthManthan, @buddyjojo Please check that the changes I made look good to you and that **it works properly**, since I might have messed something up.

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add sound and modem

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add audio jack detection

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-fortunaltezt: Add accelerometer

This device has ST LIS2HH12 accelerometer.
Add support for it.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>
[Use common &st_accel definition from gprime.dtsi]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add vibrator

Motor is powered by VPH_PWR (aka Vbat) and controlled by GPIO.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add touch screen

Like msm8916-samsung-a3u-eur, the Grand Prime uses a Zinitix BT541
touch screen. Add it together with the necessary fixed-regulator to the
device tree.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add rt5033 battery

Like the Samsung Galaxy A3/A5, the Grand Prime uses a Richtek RT5033 PMIC
as battery fuel gauge, charger, flash LED and for some regulators.
For now, only add the fuel gauge/battery device to the device tree, so we
can check the remaining battery percentage.

The other RT5033 drivers need some more work first before they can be used
properly.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add display panel

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-fortunaltezt: Add NFC

Grand Prime SM-G530Y (fortunaltezt) has a NXP PN547, which is supported
by the nxp-nci-i2c driver in mainline. It seems to detect NFC tags
using "nfctool" just fine, although more testing is difficult given
there seem to be very few useful applications making use of the
Linux NFC subsystem. :(

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>
[Move i2c-nfc to gprime.dtsi to share it with other variants]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

buddyjojo committed:
arm64: dts: qcom: msm8916-samsung-gprimeltecan: Add accelerometer/magnetometer

Grand Prime Canada uses a Bosch BMC150 accelerometer/magnetometer combo.
The chip provides two separate I2C devices for the accelerometer
and magnetometer that are already supported by the bmc150-accel
and bmc150-magn driver

Signed-off-by: Joe Mason <buddyjojo06@outlook.com>
[Move sensors to gprime.dtsi, disabled by default]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

SiddharthManthan committed:
arm64: dts: qcom: msm8916-samsung-fortuna3g: Add initial device tree

Signed-off-by: Siddharth Manthan <siddharth.manthan@gmail.com>

SiddharthManthan committed:
arm64: dts: qcom: msm8916-samsung-fortuna3g: Add accelerometer/magnetometer

This device has ST LSM303C accelerometer/magnetometer combo.
Add support for it.

Signed-off-by: Siddharth Manthan <siddharth.manthan@gmail.com>
[Move sensors to gprime.dtsi (disabled by default)]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

SiddharthManthan committed:
arm64: dts: qcom: msm8916-samsung-fortuna3g: Add PWM backlight

Signed-off-by: Siddharth Manthan <siddharth.manthan@gmail.com>
[Move clk_pwm_backlight to gprime.dtsi and disable only on fortunaltezt]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprime: Add common device tree

The Samsung Galaxy Grand Prime is a phone based on MSM8916. It is similar
to the other Samsung devices based on MSM8916 with only a few minor
differences.

The device tree contains initial support for the Grand Prime with:
 - UART
 - eMMC/SD card
 - Buttons
 - WiFi/Bluetooth (WCNSS)
 - USB

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-gprimeltecan: Add initial device tree (v5)

v5: Rename from msm8916-samsung-gprime

Samsung Galaxy Grand Prime (SM-G530W) is a smartphone using the MSM8916
SoC released in 2014.

The Grand Prime variants are very similar, with some differences in
accelerometer and NFC. The common parts are shared in
msm8916-samsung-gprime-common.dtsi to reduce duplication.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-fortunaltezt: Add initial device tree

Samsung Galaxy Grand Prime (SM-G530Y) is a smartphone using the MSM8916
SoC released in 2014.

The Grand Prime variants are very similar, with some differences in
accelerometer and NFC. The common parts are shared in
msm8916-samsung-gprime-common.dtsi to reduce duplication.

Unfortunately, SM-G530Y was released with outdated 32-bit only
firmware and never received any update from Samsung. Since the 32-bit
TrustZone firmware is signed there seems to be no way currently to
actually boot this device tree on arm64 Linux at the moment. :(

However, it is possible to use this device tree by compiling an ARM32 kernel
instead. The device tree can be easily built on ARM32 with an #include
and it works really well there. To avoid confusion for others it is still
better to add this device tree on arm64. Otherwise it's easy to forget
to update this one when making some changes that affect all MSM8916 devices.

Maybe someone finds a way to boot ARM64 Linux on this device at some point.
In this case I expect that this device tree can be simply used as-is.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>

bolilingmeng89006 committed:
ARM: dts: qcom-msm8916-samsung-fortunaltezt: Include dts from arm64

After adding all necessary support for MSM8916 SMP/cpuidle without PSCI
on ARM32, build the Samsung Galaxy Grand Prime device tree from the arm64
tree together with the ARM32 include to allow booting this device on ARM32.

The approach to include device tree files from other architectures is
inspired from e.g. the Raspberry Pi (bcm2711-rpi-4-b.dts) where this is
used to build the device tree for both ARM32 and ARM64.

Signed-off-by: Lin, Meng-Bo <linmengbo0689@protonmail.com>
[Add qcom,srot-locked to workaround TSENS problem]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

buddyjojo committed:
arm64: dts: qcom: msm8916-samsung-gprimeltecan: Add NFC

The Samsung Galaxy Grand Prime CAN has a Samsung S3FWRN5 NFC chip that
works quite well with the s3fwrn5 driver in the Linux NFC subsystem.

The clock setup for the NFC chip is a bit special (although this
seems to be a common approach used for Qualcomm devices with NFC):

The NFC chip has an output GPIO that is asserted whenever the clock
is needed to function properly. On the A3/A5 this is wired up to
PM8916 GPIO2, which is then configured with a special function
(NFC_CLK_REQ or BB_CLK2_REQ).

Enabling the rpmcc RPM_SMD_BB_CLK2_PIN clock will then instruct
PM8916 to automatically enable the clock whenever the NFC chip
requests it. The advantage is that the clock is only enabled when
needed and we don't need to manage it ourselves from the NFC driver.

Signed-off-by: Joe Mason <buddyjojo06@outlook.com>
[Put NFC pinctrl into gprime.dtsi to share it with other variants]
Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

stephan-gh committed:
thermal: qcom: tsens: Add workaround for locked SROT region

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

stephan-gh committed:
iio: accel: st_accel: Add LSM303C

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

stephan-gh committed:
iio: magnetometer: st_magn: Add LSM303C

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

Commit:
squash! MSM8916: drm/panel: Generate using linux-mdss-dsi-panel-driver-generator

X-Code-Generator: msm8916-mainline/linux-panel-drivers@3b139bb
Signed-off-by: lmdpdg <lmdpdg@localhost>

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-samsung-gprime.dtsi:
```diff
@@ -74,6 +87,29 @@
+		pinctrl-names = "default";
+		pinctrl-0 = <&tsp_int_default>;
+
+		linux,keycodes = <KEY_APPSELECT KEY_BACK>;
```
does the multi-touch work?

TravMurav commented on arch/arm64/boot/dts/qcom/Makefile:
```diff
@@ -28,7 +28,6 @@ dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a3u-eur.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a5-zt.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a5u-eur.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-e7.dtb
-dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-gprime.dtb
```
This commit drops the dtb and doesn't actually introduce anything. Is it supposed to be squashed with some other commits maybe? I'd assume that an introductory commit for similar devices would look like this: https://github.com/msm8916-mainline/linux/commit/45f778028c29d144ff27e86aa63bb49b1b2cf418

wonderfulShrineMaidenOfParadise replied:
https://github.com/msm8916-mainline/linux/pull/228/commits/00c01dcc6f589fe75eff94fa03b14a27a3e2174b#diff-6ef2edc037b112b898a8ae8cf2e368a6cbc75dda5cc8a5dc9be6b1146ca4569dR32
https://github.com/msm8916-mainline/linux/pull/228/commits/2661127949f6f37c1c1e1a8b3025c58a28137b81#diff-6ef2edc037b112b898a8ae8cf2e368a6cbc75dda5cc8a5dc9be6b1146ca4569dR31
https://github.com/msm8916-mainline/linux/pull/228/commits/b867e93b06e82bb473aeb502d8326a9fb7b1faaf#diff-6ef2edc037b112b898a8ae8cf2e368a6cbc75dda5cc8a5dc9be6b1146ca4569dR31

New dtb will be added separately in 3 init commits.

stephan-gh replied:
I'll squash this one probably somehow but for authorship I think it's better to add the dtbs separately.

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-samsung-gprimeltecan.dts:
```diff
@@ -0,0 +1,29 @@
+			no-map;
+		};
+
+		mpss_mem: mpss@86800000 {
```
I'd then prefer to have the modem mem addition in the modem commits and all other stuff (model and tz_apps) in the initial commit maybe?

stephan-gh replied:
I think we might be able to upstream modem stuff soon so I'd keep it in one commit for now.

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-samsung-fortuna3g.dts:
```diff
@@ -0,0 +1,29 @@
+			no-map;
+		};
+
+		mpss_mem: mpss@86800000 {
```
same as the gprimeltecan

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-samsung-fortunaltezt.dts:
```diff
@@ -0,0 +1,41 @@
+			no-map;
+		};
+
+		mpss_mem: mpss@86800000 {
```
same as other two

TravMurav commented on arch/arm/boot/dts/qcom-msm8916-samsung-fortunaltezt.dts:
```diff
@@ -0,0 +1,7 @@
+#include "qcom-msm8916-smp.dtsi"
+
+&tsens {
+	qcom,srot-locked;
```
maybe have a comment here?

stephan-gh committed:
wcn36xx: txrx: Ignore 5 GHz scan packets on WCN3620 for now

It's not clear why it happens but some WCN3620 firmwares seem to
occasionally seems to send scan packets with rf_band = 0, which is
currently interpreted like 5 GHz band. This causes big WARNINGs later
because those bands are not initialized for WCN3620:

WARNING: CPU: 0 PID: 0 at net/mac80211/rx.c:4877 ieee80211_rx_list+0x488/0xa10 [mac80211]
Hardware name: Samsung Galaxy Grand Prime (SM-G530H) (DT)
pc : ieee80211_rx_list+0x488/0xa10 [mac80211]
lr : ieee80211_rx_napi+0x58/0xe0 [mac80211]
Call trace:
 ieee80211_rx_list+0x488/0xa10 [mac80211]
 ieee80211_rx_napi+0x58/0xe0 [mac80211]
 ieee80211_tasklet_handler+0xe4/0xf0 [mac80211]
 ...

Let's skip the check for WCN3620 for now and warn only once to avoid
spam in the kernel log.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>

TravMurav reviewed (approved):
assuming that the commits are properly squashed on merge/rebase, it looks good to me

wonderfulShrineMaidenOfParadise commented:
Looks good after testing on fortunaltezt.

SiddharthManthan commented:
Works well on fortuna3g

stephan-gh commented:
Pushed to rebase
